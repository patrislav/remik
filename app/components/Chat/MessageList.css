.wrapper {
  width: 100%;
  height: 100%;
}

.list {
  list-style-type: none;
  margin: 0;
  padding: 0.3rem;
}

.item {

}
