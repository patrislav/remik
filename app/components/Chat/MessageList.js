import React, {PropTypes} from 'react'
import styles from './MessageList.css'

const propTypes = {
  messages: PropTypes.array
}

const defaultProps = {
  messages: []
}

const MessageList = ({ messages }) => (
  <ul className={styles.list}>
    {messages.map((message, i) =>
      <li key={i} className={styles.item}>{message.author}: {message.content}</li>
    )}
  </ul>
)

MessageList.propTypes = propTypes
MessageList.defaultProps = defaultProps

export default MessageList
