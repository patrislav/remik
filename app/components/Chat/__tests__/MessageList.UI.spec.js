import React from 'react'
import {shallow} from 'enzyme'
import {shallowToJson} from 'enzyme-to-json'
import times from 'lodash/times'

import MessageList from '../MessageList'

describe('<MessageList />', () => {
  it('renders correct UI with default props', () => {
    const wrapper = shallow(<MessageList />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('renders correct UI when passed a list of messages', () => {
    const messages = times(3, i => ({
      author: `User #${i}`, content: `Message #${i}`, timestamp: 'TIMESTAMP'
    }))

    const wrapper = shallow(<MessageList messages={messages} />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
