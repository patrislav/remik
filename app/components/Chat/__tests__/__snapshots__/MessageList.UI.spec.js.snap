exports[`<MessageList /> renders correct UI when passed a list of messages 1`] = `
<ul
  className="list">
  <li
    className="item">
    User #0
    : 
    Message #0
  </li>
  <li
    className="item">
    User #1
    : 
    Message #1
  </li>
  <li
    className="item">
    User #2
    : 
    Message #2
  </li>
</ul>
`;

exports[`<MessageList /> renders correct UI with default props 1`] = `
<ul
  className="list" />
`;
